### PR TITLE
windows-terminal@1.13.11431.0: Add context menu

### LINKED ISSUE #7658: [Feature Request] windows-terminal: context menu entries
## Bug Report

**Package Name:** windows-terminal

### Current Behaviour
There is no context menu.

### Expected Behaviour
There is a context menu.

### Possible Solution
Add .reg file like `notepadplusplus`, `sublime-text`, etc

### System details

**Windows version:** 10

**OS architecture:** 64bit

**PowerShell version:** [output of `"$($PSVersionTable.PSVersion)"`]
```
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      19041  1320
```
**Additional software:** [(optional) e.g. ConEmu, Git]

#### Scoop Configuration

```json
{
    "lastupdate":  "2022-01-03T13:38:18.5943671+07:00",
    "SCOOP_REPO":  "https://github.com/ScoopInstaller/Scoop/",
    "SCOOP_BRANCH":  "master",
    "MSIEXTRACT_USE_LESSMSI":  true
}
```

I want to try but I dont really understand about windows registry, so I'm afraid I will messed up my registry.

### PR DESCRIPTION
**Pull request by nagromc**
Closes #7658

Relates to #8636 

- [X] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).

### CODE REVIEW

File: bucket/windows-terminal.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "description": "The new Windows Terminal, and the original Windows console host - all in the same place!",
     "homepage": "https://github.com/microsoft/terminal",
     "license": "MIT",
+    "notes": "Add Windows Terminal as a context menu option by running as an administrator `reg import \"$dir\\install-context.reg\"`",
```
Comment (rashil2000) [inline bucket/windows-terminal.json:6]: Admin rights are not required for HKEY_CURRENT_USER, right?

Comment (nagromc) [reply bucket/windows-terminal.json:6]: Well, on my machine, the command won’t succeed unless I run the command `reg import` as an administrator. Maybe I miss something.

Would you please try to run the command as a regular user, and then tell me if you manage to add the context menu?

Comment (nagromc) [reply bucket/windows-terminal.json:6]: The [official documentation](https://docs.microsoft.com/en-us/windows-server/administration/windows-commands/reg-import) says:

> Copies the contents of a file that contains exported registry subkeys, entries, and values into the registry of the **local computer**.

To be honest, I copy-paste-adapted the logic from [Sublime Text](https://github.com/ScoopInstaller/Extras/blob/master/bucket/sublime-text.json) and [Notepad++](https://github.com/ScoopInstaller/Extras/blob/master/bucket/notepadplusplus.json).

Comment (nagromc) [reply bucket/windows-terminal.json:6]: Oh, I think I got it: https://stackoverflow.com/questions/59917614/edit-the-windows-explorer-context-menu-for-only-a-single-user

I’ll try that out.

Comment (nagromc) [reply bucket/windows-terminal.json]: Commit [8fe7396](https://github.com/ScoopInstaller/Extras/pull/8637/commits/8fe7396df251f6015db795000c8b29497dccd459) should do the trick.

**Review by rashil2000 — APPROVED** [review]

File: bucket/windows-terminal.json [hunk 2]
```diff
@@ -19,11 +20,21 @@
...
+    "post_install": [
+        "'install-context', 'uninstall-context' | ForEach-Object {",
+        "    if (Test-Path \"$bucketsdir\\extras\\scripts\\windows-terminal\\$_.reg\") {",
+        "        $wtPath = \"$dir\\wt.exe\".Replace('\\', '\\\\')",
+        "        $content = (Get-Content \"$bucketsdir\\extras\\scripts\\windows-terminal\\$_.reg\").Replace('$wt', $wtPath)",
+        "        if ($global) { $content = $content.Replace('HKEY_CURRENT_USER', 'HKEY_LOCAL_MACHINE') }",
+        "        Set-Content \"$dir\\$_.reg\" $content -Encoding Ascii -Force",
+        "    }",
+        "}"
+    ],
```
Comment (issaclin32) [inline bucket/windows-terminal.json:37]:
```suggestion
    ],
    "uninstaller": {
        "script": [
            "if ($cmd -eq 'uninstall') {",
            "    reg import \"$dir\\uninstall-context.reg\"",
            "}"
        ]
    },
```

Comment (issaclin32) [reply bucket/windows-terminal.json:37]: I think many users would **forget to unregister the context menu** before uninstalling, which can cause some trouble. This **uninstaller** script avoids the problem.

"uninstall-context" will only execute when the user runs `scoop uninstall ...`, but not `scoop update ...`

Comment (nagromc) [reply bucket/windows-terminal.json:37]: This is a good idea. I was surprised [Sublime Text](https://github.com/ScoopInstaller/Extras/blob/master/bucket/sublime-text.json) and [Notepad++](https://github.com/ScoopInstaller/Extras/blob/master/bucket/notepadplusplus.json) didn’t implemented such a feature (I can’t see any `uninstaller` field on those manifests).

I will add it to the manifest soon.

Comment (issaclin32) [reply bucket/windows-terminal.json:37]: > I was surprised Sublime Text and Notepad++ didn’t implemented such a feature

Thanks for telling me. I will add this to Sublime Text and Notepad++ later.

**Pull request merged**